bin/improver-spot-extract

```diff
@@ -122,6 +153,52 @@ def main():
         grid_metadata_identifier=args.grid_metadata_identifier)
     result = plugin.process(neighbour_cube, diagnostic_cube)
 
+    # If a probability or percentile diagnostic cube is provided, extract
+    # the given percentile if available. This is done after the spot-extraction
+    # to minimise processing time; usually there are far fewer spot sites than
+    # grid points.
+    if args.extract_percentile:
+        try:
+            perc_coordinate = find_percentile_coordinate(result)
+        except CoordinateNotFoundError:
+            if 'probability_of_' in result.name():
+                result = GeneratePercentilesFromProbabilities(
+                    ecc_bounds_warning=args.ecc_bounds_warning).process(
+                        result, percentiles=args.extract_percentile)
+                result = iris.util.squeeze(result)
+            elif result.coords('realization', dim_coords=True):
+                fast_percentile_method = (
+                    False if np.ma.isMaskedArray(result.data) else True)
+                result = PercentileConverter(
+                    'realization', percentiles=args.extract_percentile,
+                    fast_percentile_method=fast_percentile_method).process(
+                        result)
+                # This ensures the output for percentiles derived from
+                # realization input looks like that derived from other inputs.
+                result.coord('percentile_over_realization').rename(
```

Comment:
Is there any reason why this shouldn't go into the plugin? Apart from causing you more work (sorry)

Reply:
Only that I don't know whether we want to permanently change this naming convention for all use cases. I didn't find Bruce's document, that Caroline linked on the ticket, very clear on the plan (did I skim it too fast?).

Reply:
No I think the document just isn't very clear yet!  My preference would be that we do this in the plugin and permanently change all use cases to 'percentile' (the '_over_X' part of the name is really just provenance metadata - it shouldn't matter where the distribution came from, the percentile is still a percentile).  We should make sure @bjwheltor and @benfitzpatrick are OK with that though.

Reply:
Revisiting this.  Even if we defer the decision on what to call the percentile coordinate, it should always be in units of '%', so that's a bug in PercentileConverter.  Can that be fixed in this PR, or do we need a follow-on ticket?

Reply:
I'll write a little follow on ticket that can cover the unit change and the name change, the latter pending agreement.

Reply:
IMPRO-1161 created.